app/models/pipeline_run.rb

```diff
@@ -276,6 +291,48 @@ def db_load_ercc_counts
     update(total_ercc_reads: ercc_counts_array.map { |entry| entry[:count] }.sum)
   end
 
+  def db_load_amr_counts
+    # test files: "s3://idseq-database/test/AMR/amr_processed_results.csv", "s3://idseq-database/test/AMR/amr_summary_results.csv"
+    amr_results = PipelineRun.download_file(s3_file_for("amr_full_results"), local_amr_full_results_path)
```

Comment:
We noticed that loading failed because amr_results was nil. Given that this function only gets called once `check_and_enqueue` detects that the S3 file is present, my only explanation is that `PipelineRun.download_file` randomly failed. My recommendation is: change it to use `download_file_with_retries`, and return early from `db_load_amr_counts` if `amr_results` is still nil. It should then keep doing `db_load_amr_counts` every couple minutes until it succeeds in downloading the file and `amr_results` is not nil. Only then should it move on to the next block.

Reply:
Try that and see how it goes.

Reply:
You'll notice that `db_load_taxon_counts` functions in the way I just described